### PR TITLE
Revert "Revert "Revert "Revert "[RLlib] Upgrade to gymnasium 1.0.0 (ale_py 0.10.1, mujoco 3.2.4, pettingzoo 1.24.3 supersuit 3.9.3).""""

### PR DESCRIPTION
**Pull request by sven1977**
Reverts ray-project/ray#48435

### CODE REVIEW

**Comment by aslonnie** [comment]
will leave this to @can-anyscale and other people. let me know if you need explicit approval on the `setup.py` file.

we might be breaking a record of the number of "revert"'s in the title..

maybe consider breaking this PR into smaller ones so that it is easier to make progress(though I am not sure what is the best way to break it)? maybe we can accept using `gymnasium ~= 1.0` in `setup.py` to have some wiggle room?

**Review by aslonnie — COMMENTED** [review]
all power to @can-anyscale

File: rllib/env/single_agent_env_runner.py
Comment (simonsays1980) [inline rllib/env/single_agent_env_runner.py:309]: Isn't there a gymnasium wrapper that does this?

File: rllib/env/single_agent_env_runner.py
Comment (simonsays1980) [inline rllib/env/single_agent_env_runner.py:627]: Very nice! I remember in my problems with the update nto gymnasium==1.0.0 this was one reason of failure. It needs now this mode.

**Review by simonsays1980 — APPROVED** [review]
LGTM. Huge change now. Maybe I overread it, but could you change the environment also in the BC benchmark with RLUnplugged?

File: rllib/env/multi_agent_env_runner.py
Comment (sven1977) [reply rllib/env/multi_agent_env_runner.py:266]: This is only RLlib-internally b/c the vector env is wrapped to return info-lists (rather than dicts).
I think we still have to unify the way the user sees these envs in their callbacks: Always as-is (vector wrapper) or always as unwrapped (pure gym.vector.Env). Both approaches have advantages and disadvantages.

**Review by can-anyscale — APPROVED** [review]

**Comment by sven1977** [comment]
Hey @aslonnie , could we give this a go? Release tests are passing properly now:
https://buildkite.com/ray-project/release/builds/25039

**Review by aslonnie — APPROVED** [review]

**Review by richardliaw — APPROVED** [review]

**Pull request merged**
